include/swift/PrintAsClang/ClangMacros.def

```diff
@@ -0,0 +1,234 @@
+//===--- ClangMacros.def - Macros emitted by PrintAsClang ----=--*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the database of macros emitted into compatibility headers
+// generated by PrintAsClang.
+//
+//===----------------------------------------------------------------------===//
+
+// CLANG_MACRO(NAME, ARGS, VALUE)
+// A simple name/value macro that does not depend on a condition.
+#ifndef CLANG_MACRO
+#define CLANG_MACRO(NAME, ARGS, VALUE)
+#endif
+
+// CLANG_MACRO_BODY(NAME, BODY)
+// A macro conditionalized on the name, but that defines a helper macro or
+// other kind of additional definitions other than the macro itself.
+#ifndef CLANG_MACRO_BODY
+#define CLANG_MACRO_BODY(NAME, BODY)
+#endif
+
+// CLANG_MACRO_DEFINED(NAME)
+// Used to signal that a macro is defined by another CLANG_MACRO_BODY call.
+// This is included so that these macros can be included in the generated
+// `compatibility_symbols` file.
+#ifndef CLANG_MACRO_DEFINED
+#define CLANG_MACRO_DEFINED(NAME)
+#endif
+
+// CLANG_MACRO_ALTERNATIVE(NAME, ARGS, CONDITION, VALUE, ALTERNATIVE)
+// A simple name/value macro that evaluates to the given value when the given
+// condition is true, or to the alternative value when it is false.
+#ifndef CLANG_MACRO_ALTERNATIVE
+#define CLANG_MACRO_ALTERNATIVE(NAME, ARGS, CONDITION, VALUE, ALTERNATIVE)
+#endif
+
+// CLANG_MACRO_CONDITIONAL(NAME, ARGS, CONDITION, VALUE)
+// A simple name/value macro that only evaluates to the given value when the
+// given condition is true. Otherwise the macro will evaluate to nothing.
+#ifndef CLANG_MACRO_CONDITIONAL
+#define CLANG_MACRO_CONDITIONAL(NAME, ARGS, CONDITION, VALUE) \
+    CLANG_MACRO_ALTERNATIVE(NAME, ARGS, CONDITION, VALUE, )
+#endif
+
+// CLANG_MACRO_OBJC(NAME, ARGS, VALUE)
+// A simple name/value macro that is only defined when being compiled as
+// Objective-C.
+#ifndef CLANG_MACRO_OBJC
+#define CLANG_MACRO_OBJC(NAME, ARGS, VALUE)
+#endif
+
+// CLANG_MACRO_CXX(NAME, ARGS, VALUE, ALTERNATIVE)
+// A simple name/value macro that evaluates to the given value when being
+// compiled as C++, or as the alternative when not.
+#ifndef CLANG_MACRO_CXX
+#define CLANG_MACRO_CXX(NAME, ARGS, VALUE, ALTERNATIVE)
+#endif
+
+// CLANG_MACRO_CXX_BODY(NAME, BODY)
+// A complex macro conditionalized on whether the source is being compiled as
+// C++.
+#ifndef CLANG_MACRO_CXX_BODY
+#define CLANG_MACRO_CXX_BODY(NAME, BODY)
+#endif
+
+CLANG_MACRO_DEFINED("SWIFT_TYPEDEFS")
+
+#define MAP_SIMD_TYPE(C_TYPE, SCALAR_TYPE, _) \
```

Comment:
These aren't macros but rather typedefs. Would it make more sense for this logic to go into the actual executable.

Reply:
I feel like the purpose of this file is mainly to accumulate the symbols that PrintAsClang adds to the compatibility header, whether they're macros or not. The infrastructure treats them all as macros, but the end result is the same: We can use the resulting list in clang-extractapi to filter out the symbols that didn't actually come from user code.